Switch/SWCoreWindowService.m

```diff
@@ -373,8 +368,11 @@ - (void)private_displayTimerFired:(NSTimer *)timer;
         return;
     }
 
-    [self.stateMachine displayTimerCompleted];
-    self.displayTimer = nil;
+    // Do not show the interface if the state machine has an action pending.
+    if (!self.stateMachine.pendingSwitch) {
+        [self.stateMachine displayTimerCompleted];
+        self.displayTimer = nil;
+    }
```

Comment:
This stuff should be factored into the state machine. It's conditional on the state machine's state, and is intended to modify a state machine action (displaying the interface).